src/transforms.jl

```diff
@@ -111,27 +152,53 @@ function transform_slots(expr, symbols)
   expr
 end
 
-"""
-Function that handles `let` block
-"""
-function transform_slots_let(expr::Expr, symbols)
-  @capture(expr, let vars_; body_ end)
-  locals = Set{Symbol}()
-  (isa(vars, Expr) && vars.head==:(=))  || error("@resumable currently supports only single variable declarations in let blocks, i.e. only let blocks exactly of the form `let i=j; ...; end`. If you need multiple variables, please submit an issue on the issue tracker and consider contributing a patch.")
-  sym = vars.args[1].args[2].value
-  push!(locals, sym)
-  vars.args[1] = sym
-  body = postwalk(x->transform_let(x, locals), :(begin $(body) end))
-  :(let $vars; $body end)
+#"""
+#Function that handles `let` block
+#"""
+#function transform_slots_let(expr::Expr, symbols)
+#  @capture(expr, let vars_; body_ end)
+#  locals = Set{Symbol}()
+#  (isa(vars, Expr) && vars.head==:(=))  || error("@resumable currently supports only single variable declarations in let blocks, i.e. only let blocks exactly of the form `let i=j; ...; end`. If you need multiple variables, please submit an issue on the issue tracker and consider contributing a patch.")
+#  sym = vars.args[1].args[2].value
+#  push!(locals, sym)
+#  vars.args[1] = sym
+#  body = postwalk(x->transform_let(x, locals), :(begin $(body) end))
+#  :(let $vars; $body end)
+#end
+
+function transform_let(expr)
+  expr isa Expr || return expr
+  expr.head === :block && return expr
+  #@info "inside transform let"
+  @capture(expr, let arg_; body_; end) || return expr
+  #@info "captured let"
+  #arg |> dump
+  #@info expr
+  #@info arg
+  #error("ASds")
+  res = quote
+    let
+      local $arg
+      $body
+    end
+  end
+  #@info "emitting $res"
+  res
+  #expr.head === :. || return expr
+  #expr = expr.args[2].value in symbols ? :($(expr.args[2].value)) : expr
 end
 
 """
 Function that replaces a variable `_fsmi.x` in an expression by `x` where `x` is a variable declared in a `let` block.
 """
-function transform_let(expr, symbols::Set{Symbol})
+function transform_local(expr)
```

Comment:
Just curious about the naming convention. If this is specifically about let blocks, why do you prefer calling it `_local`? I am not asking for this to be changed, rather I just have easier time understanding other people's code if I understand the reasoning behind names.

Reply:
gone